ui/app/components/sender-to-recipient/sender-to-recipient.component.js

```diff
@@ -38,11 +38,17 @@ export default class SenderToRecipient extends PureComponent {
     recipientAddressCopied: false,
   }
 
+  recipientAddress = this.props.recipientAddress
+    ? checksumAddress(this.props.recipientAddress)
+    : this.props.recipientAddress
+
+  senderAddress = checksumAddress(this.props.senderAddress)
```

Comment:
Using a class property for this would be a new pattern for us. Do you see any benefit to this over just applying the `checksumAddress` call in the parent?

Reply:
The benefit would be that we don't currently have a way for a component to document/verify/express that it requires a pre-checksummed addressed (e.g. a custom prop type validator). It makes sense to me that we would checksum the address for display purposes.

That said, we could cache this result in a bunch of other ways—would keeping it in the component state be more conventional?

Reply:
Depends on the convention :stuck_out_tongue: 

I like, and have recently been trying to follow, the convention of not manipulating data to be rendered in components. So the data the component receives is what it renders. In this case, this would require checksumming the address in `/home/danjm/kyokan/metamask-extension/ui/app/components/pages/confirm-transaction-base/confirm-transaction-base.container.js`

Why do we need to cache the result?

Reply:
> Why do we need to cache the result?

If we checksum the address at all it's better to that once?